rascal-lsp/src/main/rascal/library/util/LanguageServer.rsc

```diff
@@ -604,6 +604,29 @@ the right ((DocumentEdit))s immediately.
 }
 data Message(list[CodeAction] fixes = []);
 
+@synopsis{A ((analysis::diff::edits::TextEdits::TextEdit)) with additional context for LSP.}
+@description{
+In LSP, text edits can contain extra information w.r.t. ((analysis::diff::edits::TextEdits::TextEdit)).
+* label: Human-readable string that describes the change.
+* description: Human-readable string that additionally describes the change, rendered less prominently.
+* needsConfirmation: Flags whether the user should confirm this change. By default, this is false, which means that ((util::LanguageServer::TextEdit))s are applied without user information.
+
+Typically, clients provide options to group edits by label/description when showing them to the user.
+See the [LSP documentation](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#changeAnnotation) for more details.
+
+Note: to easily annotate all text edits in a ((analysis::diff::edits::TextEdits::FileSystemChange)), use the convenience keywords on ((util::LanguageServer::FileSystemChange)).
+}
+@pitfalls{
+When `needsConfirmation = false` for all edits, the client will typically apply them without showing any information from the annotations to the user.
```

Comment:
I think it has an influence on labels in undo stack?

Reply:
It does!
<img width="494" height="148" alt="afbeelding" src="https://github.com/user-attachments/assets/0da3c735-05a6-44a3-8ed7-c1c03447b337" />

Will add that as well.